app/models/edition/timestamps.rb

```diff
@@ -0,0 +1,58 @@
+class Edition::Timestamps
+  def self.edited(
+    edition,
+    payload,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    edition.temporary_first_published_at = previous_live_version&.first_published_at
+    edition.temporary_last_edited_at = now
+    edition.publisher_first_published_at = payload[:first_published_at]
+
+    unless edition.update_type == "major"
+      edition.major_published_at = previous_live_version&.major_published_at
+    end
+
+    edition.first_published_at = payload[:first_published_at] ||
+      previous_live_version&.first_published_at
+
+    edition.last_edited_at = payload[:last_edited_at] || now
+
+    edition.public_updated_at = payload[:public_updated_at] ||
+      previous_live_version&.public_updated_at
+
+    edition.publisher_major_published_at = payload[:public_updated_at]
+    edition.publisher_published_at = nil
+    edition.publisher_last_edited_at = payload[:last_edited_at]
+
+    edition.save!
+  end
+
+  def self.live_transition(
+    edition,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    unless edition.temporary_first_published_at.present?
+      edition.temporary_first_published_at = now
+    end
+
+    if edition.update_type == "major"
+      edition.major_published_at = now
+    else
+      edition.major_published_at = previous_live_version&.major_published_at
+    end
+
+    unless edition.first_published_at.present?
+      edition.first_published_at = now
+    end
+
+    edition.published_at = now
+
+    unless edition.public_updated_at.present?
```

Comment:
I'm wondering if we've not quite got our logic matching for public_updated_at

compared to https://github.com/alphagov/publishing-api/blob/master/app/commands/v2/publish.rb#L156-L164

I think we may only want to set it in [put content](https://github.com/alphagov/publishing-api/pull/1007/files#diff-e4282e225c34744d59e36393ea78c0feR21) when a value is provided and not use the `or` condition.

I feel that currently we'd have a situation where this is no longer set automatically.

Consider:

I have document 1 with edition A
I create edition B with an update type of major
And expect when I publish it to to have a public_updated_at of now

What I think happens is:
edition B has a put content without a public_updated_at so it is given the one of edition A
edition B is then published and public_updted_at is not set to now

Does that sound like a problem we have?

Reply:
Ok so we need to update `public_updated_at` to current time if `update_type == major` or if it's not already present?

Reply:
there's two parts to this (and hopefully I'm not contradicting something I said before :pray:)

In put content we should only set public_updated_at if a user has given it to us. 

When going live we should check is there a public_updated_at. If there is not it should be set to now if update_type is major otherwise previous items.

Reply:
Ok I think I've got it - I've pushed up some changes